internal/services/compute/orchestrated_virtual_machine_scale_set_resource.go

```diff
@@ -397,6 +437,18 @@ func resourceOrchestratedVirtualMachineScaleSetCreate(d *pluginsdk.ResourceData,
 		props.Zones = &zones
 	}
 
+	upgradeMode := virtualmachinescalesets.UpgradeMode(d.Get("upgrade_mode").(string))
+	rollingUpgradePolicyRaw := d.Get("rolling_upgrade_policy").([]interface{})
+	rollingUpgradePolicy, err := ExpandVirtualMachineScaleSetRollingUpgradePolicy(rollingUpgradePolicyRaw, len(zones) > 0, false)
+	if err != nil {
+		return err
```

Comment:
Could we wrap this error?

```suggestion
		return fmt.Errorf("expanding `rolling_upgrade_policy`: %+v", err)
```

Reply:
updated. Thanks